### PR TITLE
Adding task to install github private RSA key in go-server

### PR DESCRIPTION
**Pull request by macdiesel**
@edx/pipeline-team PTAL.

This PR being merged is dependent on TE-1632 being merged/completed.

### CODE REVIEW

File: playbooks/roles/go-server/tasks/main.yml [hunk 1]
```diff
@@ -126,6 +126,38 @@
...
+    owner: "{{ GO_SERVER_USER }}"
+    group: "{{ GO_SERVER_GROUP }}"
+    mode: 0600
+    state: "{{ 'file' if known_hosts_register.stat.exists else 'touch'}}"
+  become_user: "{{ GO_SERVER_USER }}"
+  tags:
+    - git_identity
+
+- name: setup known_hosts for github.com
+  command: bash -c "ssh-keyscan -t rsa,dsa github.com 2>&1 | sort -u - ~/.ssh/known_hosts > ~/.ssh/tmp_hosts && mv ~/.ssh/tmp_hosts ~/.ssh/known_hosts"
```
Comment (macdiesel) [inline playbooks/roles/go-server/tasks/main.yml:159]: To pre answer the question this dance is done here so we ensure there is no bloat in the known_hosts file and that any duplicates are sorted out. This way the play can be run multiple times on the host without any side effects.

File: playbooks/roles/go-server/tasks/main.yml [hunk 2]
```diff
@@ -126,6 +126,38 @@
...
+- name: install ssh key for the secure repos
+  copy:
+    content: "{{ GO_GIT_KEY }}"
+    dest: ~/.ssh/id_rsa
+    mode: 0600
+  become_user: "{{ GO_SERVER_USER }}"
+  tags:
+    - git_identity
+
+- stat: path=~/.ssh/known_hosts
```
Comment (doctoryes) [inline playbooks/roles/go-server/tasks/main.yml]: Why no `name:`?

File: playbooks/roles/go-server/tasks/main.yml (hunk 1 above)
Comment (doctoryes) [inline playbooks/roles/go-server/tasks/main.yml:159]: Perhaps some shortened version of this text as a comment?

Comment (macdiesel) [inline playbooks/roles/go-server/tasks/main.yml:159]: done

File: playbooks/roles/go-server/tasks/main.yml (hunk 2 above)
Comment (macdiesel) [inline playbooks/roles/go-server/tasks/main.yml]: done

**Comment by doctoryes** [comment]
👍

**Pull request merged**